nbeats_keras/model.py

```diff
@@ -73,43 +108,48 @@ def reg(layer):
-            backcast = Lambda(trend_model,
-                              arguments={"is_forecast": False, "backcast_length": self.backcast_length,
-                                         "forecast_length": self.forecast_length})(
-                theta_b)
-            forecast = Lambda(trend_model,
-                              arguments={"is_forecast": True, "backcast_length": self.backcast_length,
-                                         "forecast_length": self.forecast_length})(
-                theta_f)
-        else:  # seasonality
-            theta_b = theta_f = reg(Dense(self.backcast_length, activation='linear', name=n('theta_f_b')))(d4)
-            # theta_b = Dense(self.backcast_length, activation='linear')(d4)
-            # theta_b and theta_f are shared even for seasonality bloc
+            theta_f = theta_b = reg(Dense(nb_poly, activation='linear', name=n('theta_f_b')))
+            backcast = Lambda(trend_model, arguments={"is_forecast": False, "backcast_length": self.backcast_length, "forecast_length": self.forecast_length})
+            forecast = Lambda(trend_model, arguments={"is_forecast": True, "backcast_length": self.backcast_length, "forecast_length": self.forecast_length})
+        else:  # 'seasonality'
+            theta_b = theta_f = reg(Dense(self.backcast_length, activation='linear', name=n('theta_f_b')))
             backcast = Lambda(seasonality_model,
-                              arguments={"is_forecast": False, "backcast_length": self.backcast_length,
-                                         "forecast_length": self.forecast_length})(theta_b)
+                              arguments={"is_forecast": False, "backcast_length": self.backcast_length, "forecast_length": self.forecast_length})
             forecast = Lambda(seasonality_model,
-                              arguments={"is_forecast": True, "backcast_length": self.backcast_length,
-                                         "forecast_length": self.forecast_length})(theta_f)
-        return backcast, forecast
-
+                              arguments={"is_forecast": True, "backcast_length": self.backcast_length, "forecast_length": self.forecast_length})
+        for l in range(self.input_dim):
+            if self.has_exog():
+                d0 = Concatenate()([x[l]] + [e[ll] for ll in range(self.exo_dim)])
+            else:
+                d0 = x[l]
+            d1_ = d1(d0)
+            d2_ = d2(d1_)
+            d3_ = d3(d2_) 
+            d4_ = d4(d3_)
+            theta_f_ = theta_f(d4_)
+            theta_b_ = theta_b(d4_)
+            backcast_[l] = backcast(theta_b_)
+            forecast_[l] = forecast(theta_f_)                      
+        
+        return backcast_, forecast_
+        
     def compile_model(self, loss, learning_rate):
         optimizer = Adam(lr=learning_rate)
         self.nbeats.compile(loss=loss, optimizer=optimizer)
```

Comment:
Technically now self.compile should work.

Reply:
Sorry !!